client/src/components/MovieForm.tsx

```diff
@@ -0,0 +1,78 @@
+import { FormEvent } from "react";
+import { Movie } from "../lib/data";
+
+type movieFormProps = {
+  onAddMovie?: (e: FormEvent<HTMLFormElement>) => void;
+  onSaveMovie?: (e: FormEvent<HTMLFormElement>, newMovieId: number) => void;
+  onCancel?: () => void;
+  movie?: Movie;
+};
+
+export function MovieForm({
+  onAddMovie,
+  onSaveMovie,
+  onCancel,
+  movie,
+}: movieFormProps) {
+  function handleSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    if (movie && onSaveMovie && onCancel) {
+      onSaveMovie(e, movie.movieId!);
+      onCancel();

```

Comment:
I'm not sure I understand the purpose of calling onCancel here?
It's called here but not after onAddMovie, so it's confusing. It seems it should either be called in both cases or in neither case.